### PR TITLE
DOC: Use correct suffixes for numeric literals.

### PR DESCRIPTION
**Pull request by ashupednekar**
Added changes to c-api.coremath.rst

### CODE REVIEW

File: doc/source/reference/c-api.coremath.rst [hunk 1]
```diff
@@ -81,7 +81,22 @@ Useful math constants
 ~~~~~~~~~~~~~~~~~~~~~
 
 The following math constants are available in npy_math.h. Single and extended
-precision are also available by adding the F and L suffixes respectively.
+precision are also available by adding the f and l suffixes respectively, with the exceptions of the following: -
```
Comment (charris) [inline doc/source/reference/c-api.coremath.rst]: This doesn't look right, you need to use rst markup.

Comment (charris) [reply doc/source/reference/c-api.coremath.rst]: I would not list all the versions either, just the bare types with with a note that they use the capitalized versions.

File: doc/source/reference/c-api.coremath.rst [hunk 2]
```diff
@@ -81,7 +81,22 @@ Useful math constants
...
+-> NPY_NZEROF
+-> NPY_INFINITYL
+-> NPY_NANL 
+-> NPY_PZEROL
+-> NPY_NZEROL
+
+/*
+ * Useful constants
+ */
+#define NPY_E         2.718281828459045235360287471352662498  /* e */
```
Comment (charris) [inline doc/source/reference/c-api.coremath.rst]: These are already documented, why are you adding them here?

File: doc/source/reference/c-api.coremath.rst [hunk 3]
```diff
@@ -81,7 +81,22 @@ Useful math constants
...
+#define NPY_LOG2E     1.442695040888963407359924681001892137  /* log_2 e */
+#define NPY_LOG10E    0.4
```
Comment (charris) [inline doc/source/reference/c-api.coremath.rst]: Wrong value here in any case.

File: doc/source/reference/c-api.coremath.rst [hunk 4]
```diff
@@ -81,7 +81,22 @@ Useful math constants
...
+-> NPY_INFINITYL
+-> NPY_NANL 
+-> NPY_PZEROL
+-> NPY_NZEROL
+
+/*
+ * Useful constants
+ */
+#define NPY_E         2.718281828459045235360287471352662498  /* e */
+#define NPY_LOG2E     1.442695040888963407359924681001892137  /* log_2 e */
```
Comment (charris) [inline doc/source/reference/c-api.coremath.rst]: Note that the documentation for `NPY_LOG2E` and `NPY_LOG10E` below is a bit confusing, as I expect far more people to be familiar with `e` than its formal name of Euler-Mascheroni constant. We should slip `e` in there somehow, maybe just use `e` and put Euler-Mascheroni in parenthesis.

**Comment by rgommers** [comment]
No follow-up to comments and doesn't really look right, so closing. Thanks all.

**Pull request closed without merging**